File: src/main/java/net/onelitefeather/stardust/command/commands/InvseeCommand.java
package net.onelitefeather.stardust.command.commands;

import cloud.commandframework.annotations.Argument;
import cloud.commandframework.annotations.CommandDescription;
import cloud.commandframework.annotations.CommandMethod;
import cloud.commandframework.annotations.CommandPermission;
import cloud.commandframework.annotations.suggestions.Suggestions;
import cloud.commandframework.context.CommandContext;
import net.kyori.adventure.text.minimessage.MiniMessage;
import net.onelitefeather.stardust.StardustPlugin;
import org.bukkit.command.CommandSender;
import org.bukkit.entity.HumanEntity;
import org.bukkit.entity.Player;
import org.bukkit.util.StringUtil;
import org.jetbrains.annotations.NotNull;

import java.util.ArrayList;
import java.util.List;

public record InvseeCommand(StardustPlugin stardustPlugin) {

    @CommandMethod("invsee <player>")
    @CommandPermission("featheressentials.command.invsee")
    @CommandDescription("Allow access to an Inventory from a Player.")
    public void onCommand(@NotNull CommandSender commandSender, @Argument(value = "player", suggestions = "players") String targetName) {

        if (!(commandSender instanceof Player player)) {
            commandSender.sendMessage(MiniMessage.miniMessage().deserialize(this.stardustPlugin.getMessage("plugin.only-player-command")));
            return;
        }

        Player target = commandSender.getServer().getPlayer(targetName);
        if (target == null) {
            commandSender.sendMessage(MiniMessage.miniMessage().deserialize(this.stardustPlugin.getMessage("plugin.player-not-found", this.stardustPlugin.getPrefix(), targetName)));
            return;
        }

        player.openInventory(target.getInventory());
    }

    @Suggestions(value = "players")
    public List<String> getPlayers(CommandContext<CommandSender> context, String input) {
        List<String> strings = this.stardustPlugin.getVisiblePlayers(context.getSender()).stream().map(HumanEntity::getName).toList();
        return StringUtil.copyPartialMatches(input, strings, new ArrayList<>(strings.size()));
    }
}

File: src/main/kotlin/net/onelitefeather/stardust/command/commands/InvseeCommand.kt
package net.onelitefeather.stardust.command.commands

import cloud.commandframework.annotations.*
import io.sentry.Sentry
import net.onelitefeather.stardust.StardustPlugin
import net.onelitefeather.stardust.extenstions.addClient
import net.onelitefeather.stardust.extenstions.miniMessage
import net.onelitefeather.stardust.extenstions.toSentryUser
import org.bukkit.entity.Player

class InvseeCommand(private val stardustPlugin: StardustPlugin) {

    @CommandMethod("invsee <player>")
    @CommandPermission("stardust.command.invsee")
    @CommandDescription("Allow access to an Inventory from a Player.")
    fun onCommand(
        player: Player,
        @Argument(value = "player") target: Player,
        @Flag(value = "enderchest") enderchest: Boolean?
    ) {

        if (enderchest == true) {
            handleEnderChest(player, target)
            return
        }

        player.openInventory(target.inventory)
    }

    private fun handleEnderChest(player: Player, target: Player) {

        try {
            val permissionMessage = stardustPlugin.i18nService.getMessage(
                "plugin.not-enough-permissions",
                *arrayOf(stardustPlugin.i18nService.getPluginPrefix())
            )

            if (!player.hasPermission("stardust.command.enderchest")) {
                player.sendMessage(miniMessage { permissionMessage })
                return
            }

            if (target == player) {
                player.openInventory(player.enderChest)
            } else {

                if (player != target && !player.hasPermission("stardust.command.enderchest.others")) {
                    player.sendMessage(miniMessage { permissionMessage })
                    return
                }

                player.openInventory(target.enderChest)
            }
        } catch (e: Exception) {
            Sentry.captureException(e) {
                it.user = target.toSentryUser()
                target.addClient(it)
            }
        }
    }
}